### ISSUE TITLE
incorrect indentation on flutter `GestureDetector` with no `on` parameters

### ISSUE THREAD

**Issue by DetachHead** [dart-lang/dart_style#1100]
```dart
class _State extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
        child: Row(
      children: const [Text("Foo")],
    ));
  }
}
```
![image](https://user-images.githubusercontent.com/57028336/156906359-275d0de6-d418-4228-b083-ac43268f7384.png)

**Comment by munificent** [comment]
This is working as intended, but I agree the output doesn't look great. The problem is that your code has a mixture of argument lists with and without trailing commas. The formatter uses the presence of a trailing comma to decide how to format an argument list, so if you apply them inconsistently, you get inconsistent output. If you put a trailing comma after the call to `Row`, you get:

```dart
class _State extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      child: Row(
        children: const [Text("Foo")],
      ),
    );
  }
}
```

(Using the presence of trailing commas to affect formatting isn't ideal, but it was best way I could figure out to accommodate trailing commas when they were added to Dart without significantly changing how most existing formatted code looks.)

**Comment by DetachHead** [comment]
@munificent i don't get how this could be considered expected behavior, that formatting is unreadable to me, i've never seen any other formatter behave this way.

> ```dart
> class _State extends State<MyHomePage> {
>   @override
>   Widget build(BuildContext context) {
>     return GestureDetector(
>       child: Row(
>         children: const [Text("Foo")],
>       ),
>     );
>   }
> }
> ```

imo it should be formatted like that regardless of whether or not there's a trailing comma after the `Row()`. it could probably just insert the missing trailing comma in this case

i've updated the issue with a more accurate title and suggested solution. would you consider re-opening this issue?

**Comment by KotlinIsland** [comment]
I agree that this is a critical defect and makes the formatter highly frustrating to use.

**Comment by munificent** [comment]
> imo it should be formatted like that regardless of whether or not there's a trailing comma after the `Row()`.

That's a valid user preference (which I don't personally disagree with). The problem is that there are millions of lines of Dart code out there with argument lists that don't have trailing commas and changing the way those are all formatted would be a massive change that I can't make lightly. (When I last investigated, about 10% of lines of Dart could would be effected by this.)

I'm interested in investigating the approach but it's a serious time commitment to implement, try to build consensus on, and (if we get consensus) roll out and migrate.

For now, I'm going to leave this issue close because the formatting behavior you get *is* deliberate and the fix to get the formatting you want is simple and intended. For the larger discussion around having the formatter unify formatting styles and insert trailing commas, see #753.

**Comment by KotlinIsland** [comment]
> That's a valid user preference

Have you seen what Black does with '[magic trailing commas](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma)'?

**Comment by KotlinIsland** [comment]
actual:
```dart
class _State extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
        child: Row(
      children: const [Text("Foo")],
    ));
  }
}
```

@munificent In this example from the OP where there is a mix of trailing commas, the indentation looks the wrong way round. I would expect it to look like:

expected:
```dart
class _State extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      child: Row(
        children: const [Text("Foo")],
    ));
  }
}
```

**Comment by munificent** [comment]
Yes, that's why I said:

> and the fix to get the formatting you want is simple and intended.

Until/if dart_style inserts trailing commas for you, then all you have to do is insert a trailing comma and you'll get the formatting you want.
